src/Form/Form.php

```diff
@@ -265,6 +265,8 @@ public function setErrors(array $errors)
      */
     public function execute(array $data): bool
     {
+        $this->_data = $data;
```

Comment:
Won't this leave potentially invalid data in the object? Could we only set the data once it has been validated?

Reply:
In the case the form is invalid, `FormContext` wouldn't be able to show previous supplied data (they were restored by request data before).

`data` is (to be considered as) the supplied data, being valid or not.

Otherwise we may split in `$_requestData` and `$_data`.

However in the current behavior we can set any kind of data with $form->setData($invalidData) and we don't know if these are good (except if $form->validate($invalidData) is executed). For this reason I considered reasonable to directly set the request data.

Reply:
Good point about `setData()` also allowing invalid data.